Ch firebase instrumentation tests #157084397

daumie opened the pull request:
#### What does this PR do?
- Moves  instrumentation tests from *Circle CI Build Server* to *Firebase*. 
- Slashes the time taken to perform instrumentation tests to approx 6 mins (Was 1approx 17mins initially)
- Gives a more  detailed report on instrumentation tests by including: _Test Results_ , _Logs_ , _Test Video_

#### Description of Task to be completed?
- Have the `instrumentation tests performed from Firebase.`
- Store test results on *GCP Storage Bucket*
- Send slack notifications when  tests  fail or pass


#### How should this be manually tested?
- When a push is made on any branch, the test reports are sent to the slack channel `#art-builds` with all the necessary information.
- Any interested party can click on the available buttons to get the indepth test  reports

![image](https://user-images.githubusercontent.com/11542388/49582904-85690480-f967-11e8-8f2d-dda1b27dcf37.png)

#### Any background context you want to provide?
Instrumentation tests run on a device or an emulator. In the background, your app will be installed and then a testing app will also be installed which will control your app, launching it and running UI tests as needed.
Instrumentation tests can be used to test none UI logic as well. They are especially useful when you need to test code that has a dependency on a context.
Reducing the time taken for instrumentation tests improves feedback loop cycle to  developers increasing feedback productivity.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/157084397

daumie committed:
Download and cache dependencies

daumie committed:
Build APK with Instrumented tests

daumie committed:
Set project ID and use the service account

daumie committed:
Send test APK to  firebase

daumie committed:
Get test  reports from the storage bucket

daumie committed:
Storing the instrumented test reports

daumie committed:
Sending  notification to slack channel on succeed

daumie committed:
Sending  notification to slack channel on fail

daumie committed:
Remove redundant Instrumentation tests

daumie committed:
Add instrumented tests to circle-ci job workflow

 - Also aligns linting to redhat yaml spec

daumie committed:
Update path to mock APK and test APK

daumie committed:
Add slack message  for instrumentation tests

AnthonyGW commented on .circleci/config.yml:
```diff
@@ -374,6 +411,9 @@ workflows:
             - checkstyle_lint
+      - instrumented_test:
+          requires:
+            - test
```
Maybe add a filter for only develop, staging and master branches

daumie replied:
Thanks @AnthonyGW , I'm on it.

daumie replied:
I have added a filter to the suggested branches

AnthonyGW commented on .circleci/config.yml:
```diff
@@ -193,43 +193,80 @@ jobs:
           command: |
-            bash <(curl -s https://codecov.io/bash)
-            bash ~/repo/.circleci/notify_slack.sh
+            echo $CIRCLE_ARTIFACTS
```
Is there any purpose to these lines for `echo $CIRCLE_ARTIFACTS`?

daumie replied:
I think that was erroneously added. I will be removing it in the next push.

daumie replied:
The lines have been removed. I think they were initially used for debugging.

AnthonyGW reviewed (approved):
I've checked the reports channel on slack. It works

MadgeKinyanjui reviewed (commented):
@daumie what does the sample report sent on slack look like?

daumie committed:
Add a filter for only develop, staging and master

- Instrumentation tests will only be running on a  merge to
  the listed branches.

daumie committed:
Remove debugging lines
